program/src/processor.rs

```diff
@@ -772,11 +881,23 @@ impl Processor {
             return Err(ProgramError::InvalidArgument);
         }
 
+        let (liquidity_market_price, collateral_market_price) = get_prices_from_oracles(
+            &liquidity.oracle,
+            &collateral.oracle,
+            liquidity_oracle_info,
+            collateral_oracle_info,
+            clock,
+        )?;
+
+        let effective_interest =
+            obligation.calc_effective_interest_amount(clock.slot, liquidity.interest)?;
+        obligation.update_interest_amount(effective_interest);
```

Comment:
These two functions are always executed together (and make sense only together, because slot must be updated when pending interest is added), maybe join them into a single function?